README.md
# Landing page La Fabrique

## Developpement


TIPS: ce repo contient un sous-module, pour cloner simplement :
`git clone --recursive url.git`

**Lancer le projet**
```sh
yarn
yarn dev
```
## Modifier vos contenus

Le dossier `/config-yml` contient l'ensemble des contenus de la landing page.

`/config-yml/commons` : contenus requis pour la landing (header / footer / navigation)

`/config-yml/legals` : contenus relatifs aux obligations RGPD.

`/config-yml/modules` : modules complémentaires développés par la communauté, voir section suivante.

## Modifier vos images

Toutes les images du dossier `/public/assets/imgs` sont accessible ensuite sur votre landing. Dans vos fichiers yml, indiquez simplement le chemin à partir de ce dossier.

## Les modules
Les modules sont des composants de la landing que vous pouvez **activer** / **désactiver** en modifiant la valeur du champ `display` dans les fichiers yaml.

|  Nom | Fonctionnalités  | Fichier yml  | Spécificités 
|---|---|---|---|
| Actualités   |  <ul><li>Bloc sur la page d'accueil (derniers articles)</li><li>Page list des actualités `/actualites`</li><li>Page pour chaque actualité</li></ul> | `articles.yml`  | [Oui](#module-actualites)  |
|  Statistiques | <ul><li>Bloc KPI sur la page d'accueil</li><li>Page de statistiques `/stats`</li></ul> | `stats.yml`  | [Oui](#module-stats)  |
| Newsletter  | <ul><li>Bloc sur la page d'accueil</li></ul>  | `newsletter.yml`   |  [Oui](#module-newsletter) |
| Onboarding  | <ul><li>Bloc sur la page d'accueil</li><li>Page formulaire d'onboarding `/onboarding`</li></ul>  | `onboarding.yml`   |  [Oui](#module-onboarding) |
| Instagram  | <ul><li>Bloc feed instagram sur la page d'accueil</li></ul>  | `instagram.yml`   |  [Oui](#module-instagram) |


#### <a name="module-actualites">Actualités</a>

La propriété `articles` du fichier yml est une list, vous pouvez ajouter des articles à la suite pour les publier.

La propriété `image` doit être le nom de l'image contenue dans le dossier `/public/assets/img/articles`.

#### <a name="module-instagram">Instagram</a>

L'application a besoin d'un token instagram pour récupérer les posts et les afficher.
Ajoutez `INSTAGRAM_TOKEN=votretokenhashé` dans les sealed-secrets. (https://socialgouv.github.io/sre-tools/ pour encrypter votre token)


#### <a name="module-stats">Statistiques</a>

La propriété `page.graphs` du fichier yml est une list, vous pouvez ajouter des graphs à la suite.
Pour chaque graph, les attributs disponibles pour la propriété `type` sont :
- lineChart ([exemple](https://recharts.org/en-US/examples/SimpleLineChart))
- barChart ([exemple](https://recharts.org/en-US/examples/SimpleBarChart))


#### <a name="module-onboarding">Onboarding</a>

Le module d'onboarding contient une page avec un formulaire personnalisable.
Une requête POST est envoyée à la route spécifiée dans `url` avec un payload dynamique en fonction des attributs spécifiés dans `fields` :

|  Champs | Requis  | Description  | Valeur par défaut 
|---|---|---|---|
| label | non | Titre du champ à afficher. | ∅ |
| name | oui | Nom de la donnée dans le payload de la requête. | ∅ |
| value | non | Valeur par défaut. | ∅ |
| type | non | Type de champ HTML. Valeurs disponibles : `text` `email` `number` | text
| col | non | Nombre de colonne (/12) pour l'affichage du champ. | 6 |
| required | non | Champ requis dans le formulaire HTML. | false |
| editable | non | Champ éditable, et donc affiché sur le formulaire pour l'utilisateur. | false

#### <a name="module-newsletter">Newsletter</a>

Le module de newsletter est un bloc sur la page d'accueil avec un champ unique pour renseigner son email.
Le nom du champ contenant l'email dans le payload de la requête `POST` peut être spécifié dans `email_fieldname`. Vous pouvez également ajouter des ensembles clé/valeur dans le payload en les spécifiant dans `hidden_fields`.

## Contribuer

Vous souhaitez créer/modifier module dans la landing page ? Quelle belle initiative!

Afin que les autres projets puissent aussi en profiter, développez dans le sous-module git vers lequel pointe le dossier `/src`

Vous trouverez toutes les indications dans le [README du repository landing-core](https://github.com/SocialGouv/landing-core).



CHANGELOG.md
# [1.11.0](https://github.com/SocialGouv/1000jours-blues/compare/v1.10.5...v1.11.0) (2023-05-23)


### Features

* ci on kontinuous ([1d2d601](https://github.com/SocialGouv/1000jours-blues/commit/1d2d601ae4515f02dd09f8a66e2c58e0ee9ec7af))
* ci on kontinuous ([a0f7bb7](https://github.com/SocialGouv/1000jours-blues/commit/a0f7bb74264ed4155e3e107a6f9400c523a3a208))

## [1.10.5](https://github.com/SocialGouv/1000jours-blues/compare/v1.10.4...v1.10.5) (2023-04-21)


### Bug Fixes

* **docker:** update base image ([#47](https://github.com/SocialGouv/1000jours-blues/issues/47)) ([30909f7](https://github.com/SocialGouv/1000jours-blues/commit/30909f71dc3b4c80b59e83cd835a0aa562804dec))

## [1.10.4](https://github.com/SocialGouv/1000jours-blues/compare/v1.10.3...v1.10.4) (2023-03-29)


### Bug Fixes

* release ([7234b9b](https://github.com/SocialGouv/1000jours-blues/commit/7234b9b60afca4a8728c96e53e0374a51524abae))

## [1.10.3](https://github.com/SocialGouv/1000jours-blues/compare/v1.10.2...v1.10.3) (2023-02-14)


### Bug Fixes

* **legal-notice:** change legal notice, field 'Directeur de la publication ([cabc47a](https://github.com/SocialGouv/1000jours-blues/commit/cabc47aa8fe1b661ca8c686578b9c072e4c40960))

## [1.10.2](https://github.com/SocialGouv/1000jours-blues/compare/v1.10.1...v1.10.2) (2023-01-16)


### Bug Fixes

* **ressources:** validité des inputs ([9e76afd](https://github.com/SocialGouv/1000jours-blues/commit/9e76afd90b8b5e07872dc3a58059d8bb76b89518))

## [1.10.1](https://github.com/SocialGouv/1000jours-blues/compare/v1.10.0...v1.10.1) (2023-01-16)


### Bug Fixes

* **ressource:** demande de contact BO ([ab95a32](https://github.com/SocialGouv/1000jours-blues/commit/ab95a3220c317422d68307fdd23430e87f4f650b))
* **ressource:** source ([1631a4c](https://github.com/SocialGouv/1000jours-blues/commit/1631a4c64b2e572d5ae51ed9f1bb4ebeef289c35))
* **ressource:** source ([e559e1c](https://github.com/SocialGouv/1000jours-blues/commit/e559e1cb1ac34853a3e1d79487fdc1fc0daa539e))
* **ressrouce:** ajouter source ([eef8c5b](https://github.com/SocialGouv/1000jours-blues/commit/eef8c5bdd293c2d92f43121cc149e58483527e24))

# [1.10.0](https://github.com/SocialGouv/1000jours-blues/compare/v1.9.5...v1.10.0) (2022-12-21)


### Bug Fixes

* **ressources:** config ([94818db](https://github.com/SocialGouv/1000jours-blues/commit/94818db670a79a09f776a696cbf4d95cf7b750a7))
* **ressources:** icon accordion ([ae685e6](https://github.com/SocialGouv/1000jours-blues/commit/ae685e6e284e4cc7b214eb42d3c82d8fd58f078d))
* **ressources:** nav ([b8cce36](https://github.com/SocialGouv/1000jours-blues/commit/b8cce361ea5321ca1a29fbfc29bf8249d3a4161f))
* **ressources:** wording ([b944fd5](https://github.com/SocialGouv/1000jours-blues/commit/b944fd558abfbe87ee9cd401a5257c1a30378bda))
* **ressources:** wording ([43469c5](https://github.com/SocialGouv/1000jours-blues/commit/43469c5669af315b7e2b39e737fa986b6b33d816))


### Features

* **ressources:** ajour de la page ([849ba9d](https://github.com/SocialGouv/1000jours-blues/commit/849ba9d432ff6cba8fa111de26c227dabfa9af83))
* **ressources:** branchement strapi pour l'envoi de la demande ([a8793b2](https://github.com/SocialGouv/1000jours-blues/commit/a8793b21c655fb63b5920b270600373a477b6dad))
* **ressources:** formulaire de contact ([d39b32f](https://github.com/SocialGouv/1000jours-blues/commit/d39b32f6b71d9da3fba221ad5d2ba72917a24d87))
* **ressources:** maj données ([a7a3725](https://github.com/SocialGouv/1000jours-blues/commit/a7a3725ccaf73abf7f18fc1450f960cf8ef7de64))
* **ressources:** maj ressources ([7518aa4](https://github.com/SocialGouv/1000jours-blues/commit/7518aa4fe255d1c52a7f77b0b1a2e699a948d976))
* **ressources:** retours de PR ([df3516c](https://github.com/SocialGouv/1000jours-blues/commit/df3516c26077b4f8185d69955c6def15096c9bf8))
* **ressources:** retours de PR ([265b604](https://github.com/SocialGouv/1000jours-blues/commit/265b604e4f5c1e110f5408245ff6ba4a2bebf5cd))
* **ressources:** retours de PR ([710fe50](https://github.com/SocialGouv/1000jours-blues/commit/710fe504cec85dcb47d45ba1bbcaa4e4946e2479))

## [1.9.5](https://github.com/SocialGouv/1000jours-blues/compare/v1.9.4...v1.9.5) (2022-11-30)


### Bug Fixes

* **nav:** navigation footer ([745750b](https://github.com/SocialGouv/1000jours-blues/commit/745750b8e9530f0688b409fb110c773f7d7970aa))

## [1.9.4](https://github.com/SocialGouv/1000jours-blues/compare/v1.9.3...v1.9.4) (2022-11-30)


### Bug Fixes

* **deps:** update dependency axios to v1 ([8b9cd0c](https://github.com/SocialGouv/1000jours-blues/commit/8b9cd0c63d6c5422763f6afeefdd510d320594e0))

## [1.9.3](https://github.com/SocialGouv/1000jours-blues/compare/v1.9.2...v1.9.3) (2022-11-22)


### Bug Fixes

* **partenaire:** ajout mairie bussy ([929038e](https://github.com/SocialGouv/1000jours-blues/commit/929038e53176fd24a6bec23c45959b44ee71c305))

## [1.9.2](https://github.com/SocialGouv/1000jours-blues/compare/v1.9.1...v1.9.2) (2022-11-18)


### Bug Fixes

* **security:** config ([361e02b](https://github.com/SocialGouv/1000jours-blues/commit/361e02b7523f9369cc187ceabb1cd7f2b49af1ed))
* **security:** config ([3f5aca0](https://github.com/SocialGouv/1000jours-blues/commit/3f5aca0f7320a2e039dae70ef93e702ec409c97e))

## [1.9.1](https://github.com/SocialGouv/1000jours-blues/compare/v1.9.0...v1.9.1) (2022-11-17)


### Bug Fixes

* **security:** config ([af0c4de](https://github.com/SocialGouv/1000jours-blues/commit/af0c4de2618f3056a5730c06e86d652410c9e148))

# [1.9.0](https://github.com/SocialGouv/1000jours-blues/compare/v1.8.0...v1.9.0) (2022-11-08)


### Features

* **partenaires:** maj ([f72fde1](https://github.com/SocialGouv/1000jours-blues/commit/f72fde191489d41bcc8b4dea77f804daa08981b6))

# [1.8.0](https://github.com/SocialGouv/1000jours-blues/compare/v1.7.3...v1.8.0) (2022-11-03)


### Bug Fixes

* **dependency:** revert @dataesr/react-dsfr ([a4027bf](https://github.com/SocialGouv/1000jours-blues/commit/a4027bf3c2df0478d7f85d30c28030538f86efed))


### Features


src/pages/ressources.js
import { Accordion, AccordionItem, Col, Row } from "@dataesr/react-dsfr";
import React, { useEffect, useState } from "react";
import { page } from "../../config-yml/modules/ressources.yml";
import { Layout } from "../components/Layout";
import { useMutation } from "@apollo/client";
import { client, EPDS_CONTACT_INFORMATION, SAVE_DEMANDE_DE_CONTACT } from "../../apollo-client";
import { Spinner } from "react-bootstrap";

const ressources = page;

export default function Ressources() {
  const PATTERN_EMAIL = "[a-zA-Z0-9._-]+@[a-zA-Z0-9._-]+.[a-zA-Z.]{2,15}"
  const PATTERN_PHONE_NUMBER = "[0-9]{10}"
  const SOURCE_NAME = "1000jblues-ressources"

  const classNameByIndex = (index) => `resources-item ${index > 0 ? "resources-item-border" : ""}`

  const ItemSitesInformation = () => (
    <div>
      {ressources.epdsSitesInformation.content.map((site, index) => (
        <div key={index}>
          {showUrl(site.url, site.url)}
          <br />
        </div>
      ))}
    </div>
  )

  const ItemTelephoneLines = () => (
    <div style={{ fontSize: 14 }}>
      <div className="resources-contact-item">
        {ressources.epdsLignes.content.map((contact, index) => (
          <div style={{ marginBottom: 30 }} key={index}>
            <div className="resources-contact-title">{contact.contactName}</div>
            <div>{contact.thematic}</div>
            <div className="font-weight-bold">{contact.openingTime}</div>
            <div style={{ display: "-webkit-inline-box" }}>
              <img
                alt=""
                height={17}
                style={{ marginRight: 10 }}
                src="/assets/imgs/resources/icone-telephone.svg"
              />
              <div className="font-weight-bold">{contact.phoneNumber}</div>
            </div>
          </div>
        ))}
      </div>
    </div>
  )

  const ItemResources = () => (
    <div>
      {ressources.epdsRessourcesPremiersMois.content.map((resource, index) => (
        <div
          className={classNameByIndex(index)}
          key={index}
        >
          <b>{resource.name}</b>
          {resource.description}
        </div>
      ))}
    </div>
  )

  const ItemContact = () => {
    const [isEnabledButton, setEnabledButton] = useState(false);
    const [isEmailValid, setEmailValid] = useState(false);
    const [isPhoneNumberValid, setPhoneNumberValid] = useState(false);

    const [isLoading, setLoading] = useState(false)
    const [sendingMessage, setSendingMessage] = useState("");

    const [emailValue, setEmailValue] = useState("");
    const [phoneNumberValue, setPhoneNumberValue] = useState("");

    const handleValidEmail = (event) => {
      const emailTarger = event.currentTarget
      setEmailValid(emailTarger.validity.valid && emailTarger.value.length > 0)
      setEmailValue(emailTarger.value)
    }
    const handleValidPhone = (event) => {
      const phoneTarger = event.currentTarget
      setPhoneNumberValid(phoneTarger.validity.valid && phoneTarger.value.length > 0)
      setPhoneNumberValue(phoneTarger.value)
    }

    useEffect(() => {
      setEnabledButton(isEmailValid || isPhoneNumberValid)
    }, [isEmailValid, isPhoneNumberValid])

    const [sendEmailContactQuery] = useMutation(EPDS_CONTACT_INFORMATION, {
      client: client,
      onCompleted: () => {
        setSendingMessage("La demande a été envoyée")
        setLoading(false)
        saveContactRequest()
      },
      onError: (err) => {
        console.error(err)
        setSendingMessage("Erreur lors de l'envoi")
        setLoading(false)
      },
    })

    const [sendSaveDemandeContactQuery] = useMutation(SAVE_DEMANDE_DE_CONTACT, {
      client: client,
      onError: (err) => {
        console.error(err)
      },
    })

    const saveContactRequest = async () => {
      await sendSaveDemandeContactQuery({
        variables: {
          widgetEpdsSource: SOURCE_NAME,
        },
      })
    }

    const sendEmailOnClick = async () => {
      setSendingMessage("")
      setLoading(true)

      await sendEmailContactQuery({
        variables: {
          prenom: `[${SOURCE_NAME}]`,
          email: emailValue,
          telephone: phoneNumberValue,
          moyen: null,
          horaires: null,
        },
      })
    }

    return (
      <div className="resources">
        <p style={{ textAlign: "justify" }}>{ressources.epdsContact.content}</p>

        <Row>
          <div className="fr-input-group resources-input">
            <label className="fr-label" for="email">
              Adresse électronique
              <span className="fr-hint-text">Format attendu : nom@domaine.fr</span>
            </label>
            <input
              className="fr-input"
              name="email"
              id="email"
              type="email"
              pattern={PATTERN_EMAIL}
              onChange={handleValidEmail}
            />
          </div>

          <div className="fr-input-group resources-input">
            <label className="fr-label" for="tel">
              Numéro de téléphone
              <span className="fr-hint-text">Format attendu : (+33) 1 22 33 44 55</span>
            </label>
            <input
              className="fr-input"
              aria-describedby="tel-1-message-error"
              id="tel"
              type="tel"
              pattern={PATTERN_PHONE_NUMBER}
              onChange={handleValidPhone}
            />
          </div>
        </Row >

        <button className="fr-btn resources-button"
          onClick={sendEmailOnClick}
          disabled={!isEnabledButton}>
          {ressources.epdsContact.button}
          {isLoading && <Spinner animation="border" size="sm" style={{ marginLeft: 10 }} />}
        </button>
        <span className="resources-sending-contact">{sendingMessage}</span>
      </div>
    )
  }

  const ItemHealthProfessionals = () => (
    <div>
      {ressources.epdsProsDeSante.content.map((item, index) => (
        <div
          className={classNameByIndex(index)}
          key={index}
        >
          <b>{item.name}</b>
          <br />
          {item.description}
          <br />
          {item.url
            ? showUrl(item.url, ressources.boutonConsulterDocument)
            : ""}
        </div>
      ))}
    </div>
  )

  const showUrl = (url, text) => (
    <a
      href={url}
      target="_blank"
      style={{ textDecoration: "underline" }}
      rel="noreferrer"
    >
      {text}
    </a>
  )

  return (
    <Layout>
      <div title="Ressources" className="py-5">
        <h1>{ressources.title}</h1>
      </div>

      <Col>
        <span className="font-weight-bold">{ressources.intro.title}</span>
        <br />
        <br />
        <p>{ressources.intro.content}</p>
      </Col>

      <Accordion className="accordion-smallscreen">
        <AccordionItem title={ressources.epdsProsDeSante.title}>
          <ItemHealthProfessionals />
        </AccordionItem>
        <AccordionItem title={ressources.epdsLignes.title}>
          <ItemTelephoneLines />
        </AccordionItem>
        <AccordionItem title={ressources.epdsSitesInformation.title}>
          <ItemSitesInformation />
        </AccordionItem>
        <AccordionItem title={ressources.epdsRessourcesPremiersMois.title}>
          <ItemResources />
        </AccordionItem>
        <AccordionItem title={ressources.epdsContact.title}>
          <ItemContact />
        </AccordionItem>
      </Accordion >

    </Layout>
  );
}

